drivers/i2c-bitbang/i2c.py

```diff
@@ -0,0 +1,128 @@
+import sys
+import utime
+from machine import Pin
+
+
+class I2C:
+
+    def __init__(self, sda, scl):
+        sda.init(Pin.IN)
+        scl.init(Pin.IN)
+        sda(0)
+        scl(0)
+
+    def sda_high(self):
+        sda.init(Pin.IN)
+
+    def sda_low(self):
+        sda.init(Pin.OUT)
+
+    def scl_high(self):
+        scl.init(Pin.IN)
+
+    def scl_low(self):
+        scl.init(Pin.OUT)
+
+    def delay(self):
+        utime.sleep_us(5)
+
+    def start(self):
+        self.sda_low()
+        self.delay()
+        self.scl_low()
+        self.delay()
+
+    def restart(self):
+        self.sda_high();
+        self.delay();
+        self.scl_high();
+        self.delay();
+        self.start();
+
+    def stop(self):
+        self.sda_low();
+        self.scl_high();
+        self.delay();
+        self.sda_high();
+        self.delay();
+
+    def readbit(self):
+        self.sda_high();
+        self.scl_high();
+        self.delay();
+        bit = sda();
+        self.scl_low();
+        return bit
+
+    def writebit(self, val):
+        if val:
+            self.sda_high();
+        else:
+            self.sda_low();
```

Comment:
I don't understand such a point. I can understand if you say they **can** be replaced. That's true. And what I say is that they can be NOT replaced. And I don't see any contradiction here, or why you think your way is infallibly better.